apps/meteor/server/modules/watchers/lib/messages.ts

```diff
@@ -19,6 +19,19 @@ export async function getMessageToBroadcast({ id, data }: { id: IMessage['_id'];
 		return;
 	}
 
+	if (message.t) {
+		const hideSystemMessage = (await getSettingCached('Hide_System_Messages')) as MessageTypesValues[];
+
+		if (hideSystemMessage) {
+			const isMutedUnmuted = (messageType: string): boolean => {
+				return messageType === 'user-muted' || messageType === 'user-unmuted';
+			};
```

Comment:
Move this to outside of the func, no need to recreate it everytime `getMessageToBroadcast` is called